arkouda/io.py

```diff
@@ -977,6 +1104,84 @@ def to_hdf(
             mode = "append"
 
 
+def to_csv(
+    columns: Union[Mapping[str, pdarray], List[pdarray]],
+    prefix_path: str,
+    names: List[str] = None,
+    col_delim: str = ",",
+    overwrite: bool = False,
+):
+    """
+    Write Arkouda object(s) to CSV file(s). All CSV Files written by Arkouda
+    include a header denoting data types of the columns.
+
+    Parameters
+    -----------
+    columns: Mapping[str, pdarray] or List[pdarray]
+        The objects to be written to CSV file. If a mapping is used and `names` is None
+        the keys of the mapping will be used as the dataset names.
+    prefix_path: str
+        The filename prefix to be used for saving files. Files will have _LOCALE#### appended
+        when they are written to disk.
+    names: List[str] (Optional)
+        names of dataset to be written. Order should correspond to the order of data
+        provided in `columns`.
+    col_delim: str
+        Defaults to ",". Value to be used to separate columns within the file.
+        Please be sure that the value used DOES NOT appear in your dataset.
+    overwrite: bool
+        Defaults to False. If True, any existing files matching your provided prefix_path will
+        be overwritten. If False, an error will be returned if existing files are found.
+
+    Returns
+    --------
+    None
+
+    Raises
+    ------
+    ValueError
+        Raised if all datasets are not present in all parquet files or if one or
+        more of the specified files do not exist
```

Comment:
@Ethan-DeBandi99 in my head `raised if all datasets are not present`, means this error will only be raise if every single dataset is not present. If any single dataset is present, this error will not be raised

where `Raised if not all datasets are present` means if any dataset is missing then they're not all present, so this will be raised.

maybe the nerdiest way to do this is in python hahahah
```python
>>> present = np.array([True, False, True])
# all not present
>>> all(~present)
False
# not all present
>>> not all(present)
True
```

Reply:
we could also change the wording to `raised if any datasets are not present` and save everyone the trouble trying to parse the negative

Reply:
Updated